### DISCUSSION TITLE
Number of Lines Selected

### DISCUSSION THREAD

**Discussion by goober99** [orbitalquark/textadept#215] [Q&A] [answered]
Is there a way to get the number of lines selected with Lua?

**Comment by orbitalquark** [comment, marked as answer]
There are at least two ways I can think of:

    buffer:line_from_position(buffer.selection_end) - buffer:line_from_position(buffer.selection_start)

    select(2, buffer:get_sel_text():gsub('\n', ''))

If you always want a minimum of 1, then also call `math.max(1, result)` with your result.
